packages/provider-elasticsearch/src/example-types/results/highlighting/util.js

```diff
@@ -0,0 +1,73 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+
+export const getArrayFieldsPaths = _.memoize(
+  (schema) =>
+    _.keys(
+      _.pickBy({ elasticsearch: { meta: { subType: 'array' } } }, schema.fields)
+    ),
+  _.get('elasticsearch.index')
+)
+
+/**
+ * Returns an array of [start, end] ranges that correspond to substrings
+ * enclosed in pre/post tags. The ranges correspond to the plain string without
+ * tags. For example given the tags `{ pre: '<em>', post: '</em>' }`, this
+ * function will return [[2, 5], [6, 9]] for the string
+ *
+ * `A <em>red</em> <em>car</em>`
+ */
+const getHighlightRanges = (pre, post, str) => {
```

Comment:
Is there a reason we couldn't use `F.postings` or `F.postingsForWords`?

Reply:
`postings` is pretty different since it returns postings for the entire regex match instead of the capture

```javascript
const str = 'The <em>lazy</em> fox'
F.postings(new RegExp('<em>(.*?)</em>', 'g'), str) // -> [ [ 4, 17 ] ]
getHighlightedRanges('<em>', '</em>', str) // -> [ [ 4, 8 ] ]
```

Also `postings` does not try to pretend the tags are 0 width and this function does. Notice the difference in the end positions above (17 vs 8).

Reply:
When I was doing this prior, I thought there was a good opportunity here to update postings for words wrapped in html based on an HTML tag or something like this, but this use case is much different than what those two functions do out of the box today.